osquery/tables/system/windows/logical_drives.cpp

```diff
@@ -36,45 +41,27 @@ QueryData genLogicalDrives(QueryContext& context) {
       r["size"] = "-1";
     }
 
+    int bootPartition = 0;
+    for (const auto& bootConfiguration : bootConfigurations) {
+      std::string bootDirectory;
+      bootConfiguration.GetString("BootDirectory", bootDirectory);
```

Comment:
Can we preprocess `bootConfigurations` before outer loop? Let's say transform it to the set of device ids and check whether device id of a logical disk belongs to that set. It would be way faster. What do you think?

Reply:
Sounds good! I'll make those changes.